MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/TurfLineDistanceActivity.java

```diff
@@ -0,0 +1,215 @@
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+
+        lineLengthTextView = findViewById(R.id.line_length_textView);
+
+        // DISTANCE_UNITS must be a String found in {@link com.mapbox.turf.TurfConstants}
+        lineLengthTextView.setText(String.format(getString(R.string.line_distance_textview),
+          DISTANCE_UNITS, String.valueOf(totalLineDistance)));
+
+        TurfLineDistanceActivity.this.mapboxMap = mapboxMap;
+        mapboxMap.setStyle(new Style.Builder()
+              .fromUri("mapbox://styles/mapbox/cjv6rzz4j3m4b1fqcchuxclhb")
+
+              // Add the source to the map
+              .withSource(new GeoJsonSource(SOURCE_ID))
+
+              // Style and add the CircleLayer to the map
+              .withLayer(new CircleLayer(CIRCLE_LAYER_ID, SOURCE_ID).withProperties(
+                circleColor(CIRCLE_COLOR),
+                circleRadius(CIRCLE_RADIUS)
+              ))
+
+            // Style and add the LineLayer to the map. The LineLayer is placed below the CircleLayer.
+            .withLayerBelow(new LineLayer(LINE_LAYER_ID, SOURCE_ID).withProperties(
+              lineColor(LINE_COLOR),
+              lineWidth(LINE_WIDTH),
+              lineJoin(LINE_JOIN_ROUND)
+            ), CIRCLE_LAYER_ID), new Style.OnStyleLoaded() {
+              @Override
+              public void onStyleLoaded(@NonNull Style style) {
+                TurfLineDistanceActivity.this.mapboxMap.addOnMapClickListener(TurfLineDistanceActivity.this);
+                Toast.makeText(TurfLineDistanceActivity.this, getString(
+                  R.string.line_distance_tap_instruction), Toast.LENGTH_SHORT).show();
+              }
+            }
+        );
+      }
+    });
+  }
+
+  @Override
+  public boolean onMapClick(@NonNull LatLng point) {
+    Point clickPoint = Point.fromLngLat(point.getLongitude(), point.getLatitude());
+    addClickPointToLine(clickPoint);
+    return true;
+  }
+
+  /**
+   * Handle the map click location and re-draw the circle and line data.
+   *
+   * @param clickPoint where the map was tapped on.
+   */
+  private void addClickPointToLine(Point clickPoint) {
+    mapboxMap.getStyle(new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        // Get the source from the map's style
+        GeoJsonSource source = style.getSourceAs(SOURCE_ID);
+        if (source != null) {
+          pointList.add(clickPoint);
+
+          int listSize = pointList.size();
+
+          double distanceBetweenLastAndSecondToLastClickPoint = 0;
+
+          // Make the Turf calculation between the first and second map tap point.
+          // Add that distance to the overall line distance.
+          if (pointList.size() == 2) {
+            distanceBetweenLastAndSecondToLastClickPoint = TurfMeasurement.distance(
+              pointList.get(listSize - 1), pointList.get(0));
+          } else if (pointList.size() > 2) {
```

Comment:
Shall we combine these two cases? e.g. `pointList.size() >= 2`?

Reply:
Sure. Now it's

```java
 if (pointList.size() >= 2) {
	// If the point list only has two points, make the Turf calculation between
	// the first and second map tap point. Add that distance to the overall
	// line distance. Make the Turf calculation between the last tap point and
	// the second-to-last tap point.
	
	distanceBetweenLastAndSecondToLastClickPoint = TurfMeasurement.distance(
	pointList.get(listSize - pointList.size() == 2 ? 1 : 2),
	listSize - pointList.size() == 2 ? pointList.get(0) : pointList.get(listSize - 1));
}
```